component/src/main/java/io/siddhi/extension/io/file/util/Util.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file.util;
+
+import io.siddhi.core.event.Event;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Util Class.
+ * This method used to get the fileHandlerEvent
+ */
+public class Util {
+    public static Event getFileHandlerEvent(final File file, List<String> fileObjectList, Status enumStatus) {
+        boolean listenerEventsURLValidated = false;
+        String status;
+        switch (enumStatus) {
+            case STATUS_NEW: status = "created"; break;
+            case STATUS_PROCESS: status = "modifying"; break;
+            case STATUS_DONE: status = "modifyingCompleted"; break;
+            case STATUS_REMOVE: status = "removed"; break;
+            default: throw new IllegalStateException("Unexpected value: " + enumStatus);
+        }
+        if (fileObjectList.contains(file.getAbsolutePath())) {
+            listenerEventsURLValidated = true;
+        }
+        //If the fileObjectList contains this file
+        for (String fileObjectPath : fileObjectList) {
+            File fileObject = new File(fileObjectPath);
+            if (fileObject.isDirectory()) {
+                //If a fileObject is a folder then the events for the files in the folder should thrown.
+                if (file.getAbsolutePath().startsWith(fileObjectPath)) {
+                    listenerEventsURLValidated = true;
+                }
+            }
+        }
```

Comment:
@Ajanthy we may have to put this code segment in the else{} part of `if (fileObjectList.contains(file.getAbsolutePath()))` right?
since if the exact file is not contained in the list.. we are checking whether its a directory and we should check whether the changed event is for a file in that directory right?
In the way this code has written, the for loop will run unnecessarily even after listenerEventsURLValidated is true from the `if (fileObjectList.contains(file.getAbsolutePath()))` block

Reply:
fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/bea46527dba085b2c5bcc206f49786b72192804e